packages/react-charting/src/components/Legends/Legends.base.tsx

```diff
@@ -184,24 +191,26 @@ export class LegendsBase extends React.Component<ILegendsProps, ILegendState> {
    * @param legend ILegend
    */
 
-  private _canSelectOnlySingleLegend = (legend: ILegend): void => {
-    if (this.state.selectedLegend === legend.title) {
-      this.setState({ selectedLegend: '' });
+  private _canSelectOnlySingleLegend = (legend: ILegend): boolean => {
+    if (this.state.selectedLegends[legend.title]) {
+      this.setState({ selectedLegends: {} });
+      return false;
     } else {
-      this.setState({ selectedLegend: legend.title });
+      this.setState({ selectedLegends: { [legend.title]: true } });
+      return true;
     }
   };
 
   private _onClick = (legend: ILegend): void => {
-    if (legend.action) {
-      const { canSelectMultipleLegends = false } = this.props;
-      if (canSelectMultipleLegends) {
-        this._canSelectMultipleLegends(legend);
-      } else {
-        this._canSelectOnlySingleLegend(legend);
-      }
-      legend.action();
+    const { canSelectMultipleLegends = false } = this.props;
+    if (canSelectMultipleLegends) {
+      const selectedLegends = this._canSelectMultipleLegends(legend);
+      this.props.onChange?.(Object.keys(selectedLegends));
```

Comment:
Do you need an event or info about which legend was selected or deselected by the user

Reply:
it's in need.